src/nupic/algorithms/Connections.hpp

```diff
@@ -41,12 +41,13 @@ namespace nupic {
 namespace algorithms {
 namespace connections {
 
-typedef UInt32 CellIdx;
-typedef UInt16 SegmentIdx; /** Index of segment in cell. */
-typedef UInt16 SynapseIdx; /** Index of synapse in segment. */
-typedef UInt32 Segment;    /** Index of segment's data. */
-typedef UInt32 Synapse;    /** Index of synapse's data. */
-typedef Real32 Permanence;
+//TODO instead of typedefs, use templates for proper type-checking?
+using CellIdx   = nupic::sdr::ElemSparse; // CellIdx must match with sdr::ElemSparse, to change, change it there
```

Comment:
FYI I had to bind connections::CellIdx and sdr::ElemSparse, where ElemSparse is the element type of SDR_sparse_t (currently UInt32); 
Both have to be the same for some methods in SP, TM, so I binded them together. It could be untangeled but then I'd have to do type-conversion of the vectors in said methods.